redbot/cogs/audio/apis.py

```diff
@@ -194,12 +85,11 @@ def _make_token_auth(client_id: Optional[str], client_secret: Optional[str]) ->
             return await r.json()
 
     async def _get_auth(self):
-        if self.client_id is None or self.client_secret is None:
-            tokens = await self.bot.get_shared_api_tokens("spotify")
-            self.client_id = tokens.get("client_id", "")
-            self.client_secret = tokens.get("client_secret", "")
+        tokens = await self.bot.get_shared_api_tokens("spotify")
+        self.client_id = tokens.get("client_id", "")
+        self.client_secret = tokens.get("client_secret", "")
 
-    async def _request_token(self) -> dict:
+    async def _request_token(self) -> Mapping:
```

Comment:
`MutableMapping` should be used here instead of Mapping for the type (Same anywhere else you return a mapping which gets modified later so that dev toolings don't complain when opening the file)